src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1163,6 +1163,28 @@ export default class DefaultDeviceController
     return device && device.id ? device : null;
   }
 
+  private hasSameMediaStreamId(
+    kind: string,
+    selection: DeviceSelection,
+    proposedConstraints: MediaStreamConstraints
+  ): boolean {
+    // Checking for stream using the fake constraint created in calculateMediaStreamConstraints
+    let streamId;
+    if (kind === 'audio') {
+      // @ts-ignore
+      streamId = proposedConstraints?.audio.streamId;
+      /* istanbul ignore next */
+      // @ts-ignore
+      return streamId && streamId === selection.constraints?.audio?.streamId;
+    }
+    /* istanbul ignore next */
```

Comment:
Can we add tests to cover all scenarios?

Reply:
As discussed, it is just to be defensive and guard in all case. In reality, for video constraints for example, it is only not set if video is previously null which quit right away and if constraints is set, video property will be set.